packages/frontend/src/features/post/hooks/useReportForm/index.ts

```diff
@@ -0,0 +1,65 @@
+import { useCallback, useState } from 'react'
+import { FieldValues, useForm } from 'react-hook-form'
+
+enum ReportFormSubmitState {
+    NotSubmitted = 'not-submitted',
+    Submitting = 'submitting',
+    Success = 'success',
+    Failure = 'failure',
+}
+
+export function useReportForm(defaultValues: FieldValues) {
+    const {
+        register,
+        handleSubmit,
+        setValue,
+        getValues,
+        trigger,
+        reset,
+        formState: { errors },
+    } = useForm<FieldValues>({ defaultValues })
+
+    const [submitState, setSubmitState] = useState<ReportFormSubmitState>(
```

Comment:
Why don't you directly use formState that is returned from useForm to track submit state?
```js
// https://react-hook-form.com/docs/useform/formstate
const {
  isSubmitted,
  isSubmitting,
  isSubmitSuccessful
} = formState
```

Reply:
maintain the status quo